bin/drone/destroy-errored-agents

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+# This script will terminate all drone agents that are in 'error' state. This
+# happens occasionally, and we have not been able to pin down the exact reason
+# why. This was created when we were using the following versions, and may not
+# be needed if future bugfixes resolve this issue.
+#
+# - drone/autoscaler:1.8.2
+# - drone/drone:1.9.0
+#
+# If it's too fancy for you, you can always run the following command.
+#
+# ```
+# drone server ls -l | grep error | awk '{print $1}' | xargs -tL1 drone server destroy`
+# ```
+
+# Check if DRONE_TOKEN is valid
+output=$(drone server ls -l 2>&1)
+if [[ "$output" == *"Error: you must provide your Drone access token."* ]]; then
+    echo "DRONE_TOKEN is either not set or invalid. Follow instructions at https://drone.cdn-code.org/account to set your token."
+    exit 1
+fi
+
+# Get the total count of drone agents and those in error state
+total_count=$(drone server ls -l | tail -n +2 | wc -l)
+error_ids=$(drone server ls -l | grep error | awk '{print $1}')
+
+# Calculate error count, considering empty string as zero
+error_count=$(if [ -z "$error_ids" ]; then echo 0; else echo "$error_ids" | wc -l; fi)
```

Comment:
Will this actually work? When I try these commands locally, I get all the error ids on a single line, and this always returns a value of `1`

Reply:
This works for me:

```
> drone server ls -l
2023/09/28 14:53:21 proto: duplicate proto type registered: PluginSpec
2023/09/28 14:53:21 proto: duplicate proto type registered: PluginPrivilege
Name              Address           State      Created
agent-INK7Y0wR    54.185.23.174     error      3 days ago
agent-SULnvmZB    35.167.142.79     error      3 days ago
agent-I7G9ntrT    52.12.184.232     error      3 days ago
agent-nAQK1NQS    35.90.19.228      error      2 days ago
agent-8coN1SvG    34.219.131.79     error      2 days ago
agent-O1V6W25j    34.217.110.35     error      2 days ago
agent-ySyAREAH    54.191.222.189    error      2 days ago
agent-8LKGIE0F    34.222.139.228    error      2 days ago
agent-dmJ94UzK    34.215.20.129     error      27 hours ago
agent-FU4mWTKa    52.24.31.121      running    27 hours ago
agent-pvz72df2    34.215.210.3      error      27 hours ago
agent-2J1SPmuj    34.215.88.248     running    26 hours ago
agent-uXkj4U85    35.162.142.105    running    26 hours ago
agent-0sgjnNQz    35.88.159.185     running    26 hours ago
agent-xGUYsDRJ    52.12.6.155       running    21 hours ago


> total_count=$(drone server ls -l | tail -n +2 | wc -l)
> echo $total_count
15

> error_ids=$(drone server ls -l | grep error | awk '{print $1}')
> echo $error_ids
agent-INK7Y0wR
agent-SULnvmZB
agent-I7G9ntrT
agent-nAQK1NQS
agent-8coN1SvG
agent-O1V6W25j
agent-ySyAREAH
agent-8LKGIE0F
agent-dmJ94UzK
agent-pvz72df2

> error_count=$(if [ -z "$error_ids" ]; then echo 0; else echo "$error_ids" | wc -l; fi)
> echo $error_count
10
```

Reply:
Oh, fascinating. `$error_ids` is formatted with all the results on one line for me, but it does indeed work *as long as I wrap it in double quotes*:

```bash
$ error_ids=$(drone server ls -l | grep error | awk '{print $1}')
$ echo $error_ids
agent-INK7Y0wR agent-SULnvmZB agent-I7G9ntrT agent-nAQK1NQS agent-8coN1SvG agent-O1V6W25j agent-ySyAREAH agent-8LKGIE0F agent-dmJ94UzK agent-pvz72df2
$ error_count=$(if [ -z "$error_ids" ]; then echo 0; else echo "$error_ids" | wc -l; fi)
$ echo $error_count
10
$ echo $error_ids | wc -l
1
$ echo "$error_ids" | wc -l
10
```

Reply:
I was not aware this was a thing in bash:

```bash
$ echo "$error_ids"
agent-INK7Y0wR
agent-SULnvmZB
agent-I7G9ntrT
agent-nAQK1NQS
agent-8coN1SvG
agent-O1V6W25j
agent-ySyAREAH
agent-8LKGIE0F
agent-dmJ94UzK
agent-pvz72df2
$ echo $error_ids
agent-INK7Y0wR agent-SULnvmZB agent-I7G9ntrT agent-nAQK1NQS agent-8coN1SvG agent-O1V6W25j agent-ySyAREAH agent-8LKGIE0F agent-dmJ94UzK agent-pvz72df2

```

Reply:
trying to find docs to reference, but I believe that without quotes it's passing each 'line' as an independent argument, or it's ignoring the newline characters entirely.

Reply:
Me neither! I think this is describing the behavior? Bash docs are confusing to me :sweat_smile: 
https://www.gnu.org/software/bash/manual/html_node/Word-Splitting.html